l3kernel/l3token.dtx

```diff
@@ -2245,6 +2282,43 @@
 %    \end{macrocode}
 % \end{macro}
 %
+% \begin{macro}[pTF]{\token_if_control_word:N}
+% \begin{macro}{\@@_if_control_word_aux:w}
+%   This checks that the token consists of the escape character followed by
+%   one or more letters. As |#1| could be a control symbol let to
+%   \cs{else:} or \cs{fi:}, etc., so we use a auxiliary function
+%   to grab second \cs{if:w}.
+%    \begin{macrocode}
+\prg_new_conditional:Npnn \token_if_control_word:N #1 { p , T , F , TF }
+  {
+    \if:w 0 \tex_strcmp:D { \exp_not:N #1 } { \token_to_str:N #1 }
+      \@@_if_control_word_aux:w
+    \else:
+      \if:w \exp_not:N #1 \token_to_str:N #1
+        \prg_return_false: \else: \prg_return_true: \fi:
+    \fi:
+  }
+\cs_new:Npn \@@_if_control_word_aux:w #1 \fi: { \prg_return_false: }
+%    \end{macrocode}
+% \end{macro}
+% \end{macro}
+% \begin{macro}[pTF]{\token_if_control_symbol:N}
+%   A similar plan but we need to test for the space behavior, and then whether we
+%   are dealing with a control sequence. In this case there is no worry about
+%   a dangling token.
+%    \begin{macrocode}
+\prg_new_conditional:Npnn \token_if_control_symbol:N #1 { p , T , F , TF }
+  {
+    \if:w 0 \tex_strcmp:D { \exp_not:N #1 } { \token_to_str:N #1 }
+      \if_catcode:w \exp_not:N #1 \scan_stop:
+        \prg_return_true: \else: \prg_return_false: \fi:
+    \else:
+      \prg_return_false:
+    \fi:
+  }
```

Comment:
This breaks on input such as `\token_if_control_symbol:NTF \fi: \TRUE \FALSE` (or `\if_true:` etc.).

Better use (also gets optimised by `prg`):

```suggestion
\prg_new_conditional:Npnn \my_if_control_symbol:N #1 { p , T , F , TF }
  {
    \if_catcode:w \exp_not:N #1 \scan_stop:
    \else:
      \@@_if_control_symbol_false:w
    \fi:
    \if:w 0 \tex_strcmp:D { \exp_not:N #1 } { \token_to_str:N #1 }
      \prg_return_true:
    \else:
      \prg_return_false:
    \fi:
  }
\cs_new:Npn \@@_if_control_symbol_false:w \fi: \if:w 0 \tex_strcmp:D #1#2
  { \fi: \if_false: }
```